### PR TITLE
reduces shutdown-delay-duration to 0s for SNO

### PR DESCRIPTION
**Pull request by p0lyn0mial**
fixes https://issues.redhat.com/browse/API-1283

### CODE REVIEW

File: pkg/operator/configobservation/apiserver/observe_termination_duration.go [hunk 1]
```diff
@@ -82,8 +86,14 @@ func ObserveWatchTerminationDuration(genericListers configobserver.Listers, _ ev
...
 	}
-	switch infra.Spec.PlatformSpec.Type {
-	case configv1.AWSPlatformType:
+
+
+	switch {
+	case infra.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode:
+		// reduce termination duration from 135s (default) to 60s to reach the maximum downtime for SNO
+		// on a single node the shutdown-delay-duration is set to 0s, additional 60s is for in-flight requests
+		observedGracefulTerminationDuration = "60"
```
Comment (p0lyn0mial) [inline pkg/operator/configobservation/apiserver/observe_termination_duration.go:95]: I would actually set it to something slightly above `60s` or `135` in general

Comment (sttts) [reply pkg/operator/configobservation/apiserver/observe_termination_duration.go:94]: we have a Jira for 10s.

**Comment by p0lyn0mial** [comment]
checked https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/1164/pull-ci-openshift-cluster-kube-apiserver-operator-master-e2e-aws-single-node/1409459959950741504

from `kas-o` resource:
```
   apiServerArguments:
      ...
     shutdown-delay-duration:
      - 0s
     gracefulTerminationDuration: "60"
```

from `kas` logs `FLAG: --shutdown-delay-duration="0s"`

from the `events`:
Screenshot 2021-06-28 at 15 39 53


based on the above the new setting is respected on `SNO` platfrom

**Comment by sttts** [comment]
/retest
/lgtm
/approve

There will be a follow-up to further reduce the 60s to 15s or so when o/k got shorter in-flight timeouts during termination.

**Pull request merged**